bundles/io/org.eclipse.smarthome.io.rest.optimize/src/main/java/org/eclipse/smarthome/io/rest/optimize/internal/SmartHomeRestServletConfig.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.rest.optimize.internal;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.http.HttpContext;
+import org.osgi.service.http.HttpService;
+
+import com.eclipsesource.jaxrs.publisher.ServletConfiguration;
+
+/**
+ * Custom servlet configuration for jaxrs handler.
+ *
+ * This extension doesn't do much, just forces usage of http context created somewhere else.
```

Comment:
Yes, this is and extension point to jaxrs publisher which allows to customize OSGi servlet registration and bind it to specific http context instead of default one. Thanks to this simple hook we are able to link all jaxrs endpoints to logical ESH-specific security domain.